src/pharmpy/tools/structsearch/drugmetabolite.py

```diff
@@ -1,69 +1,105 @@
+    if (
+        model.statements.ode_system.find_compartment('METABOLITE')
+        and len(metabolite_functions) != 0
+    ):
+        raise NotImplementedError(
+            'Metabolite structure transformations not yet supported when already present in the model.'
+            ' Try removing and only applying peripheral transformations'
+        )
+    elif (
+        not model.statements.ode_system.find_compartment('METABOLITE')
+        and len(metabolite_functions) == 0
+    ):
+        raise ValueError(
+            'Require at least one metabolite model type.'
+            ' Try adding METABOLITE(BASIC) or METABOLITE(PSC) to search space'
+        )
+
+    if model.statements.ode_system.find_compartment('METABOLITE'):
+        base_description = model.description
     else:
-        presystemic_option = [False, True]
-    for presystemic in presystemic_option:
-        candidate_model = add_metabolite(model, presystemic=presystemic)
-        candidate_model = set_name(candidate_model, 'presystemic')
-
-        candidate_model = candidate_model.replace(parent_model='base_metabolite')
-        if presystemic:
-            models.append(candidate_model)
-
-        candidate_model = add_peripheral_compartment(candidate_model, "METABOLITE")
-        candidate_model = set_name(
-            candidate_model, f'{"presystemic" if presystemic else "base_metabolite"}_peripheral'
+        base_description = determine_base_description(
+            metabolite_mfl_statements, peripheral_mfl_statements
+        )
+
+    wb = WorkflowBuilder(name="drug_metabolite")
+
+    start_task = Task("start", _start, model, results)
+    wb.add_task(start_task)
+
+    def apply_transformation(eff, f, model_entry):
+        candidate_model = f(model_entry.model)
+        candidate_model = candidate_model.replace(
+            name="TEMP", description='_'.join(eff), modelfit_results=None
         )
-        if presystemic:
-            candidate_model = candidate_model.replace(parent_model='presystemic')
+        return ModelEntry.create(model=candidate_model, modelfit_results=None, parent=model)
 
-        models.append(candidate_model)
+    for eff, func in metabolite_functions.items():
+        candidate_met_task = Task(str(eff), apply_transformation, eff, func)
+        wb.add_task(candidate_met_task, predecessors=start_task)
 
-    return models
+    if len(peripheral_mfl_statements) == 0:
+        candidate_model_tasks = []
+        model_index = 1
+        for out_task in wb.output_tasks:
+            change_name_task = Task("number_run", change_name, model_index)
+            model_index += 1
+            wb.add_task(change_name_task, predecessors=[out_task])
+            wf_fit = create_fit_workflow(n=1)
+            wb.insert_workflow(wf_fit, predecessors=[change_name_task])
+
+            candidate_model_tasks += wf_fit.output_tasks
+    else:
+        wb, candidate_model_tasks = exhaustive_stepwise(
```

Comment:
Is the workflow needed as input? Perhaps it could work with something like:
```
wb_modelsearch, candidate_model_tasks = ...
wb = wb.insert_workflow(wb_modelsearch)
```

Reply:
The reason for this is because we want to apply the algorithm on multiple output nodes at the same time. The drug metabolite workflow can have two output nodes (one for `BASIC` and one for `PSC`). If the workflow were to be created and inserted for each output node, the numbering of models would overlap. Giving the workflow as input however will apply the algorithm to both nodes simultaneously, which solves this issue.

Reply:
Okay, makes sense!